File: homeWorkSeven/java/FileOperator.java
package homeWorkOne;
// Оператор, работает и импортом и экспортом фалов
import java.io.*;
import java.util.ArrayList;


public class FileOperator {

    public static void inputContactsHorisontalFormat(String fileName, PhoneBook phoneBook) {
        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
            String line = br.readLine();
            while (line != null) {
                String[] contact = line.split(" ");
                phoneBook.getAllContacts().add(new Contact(contact[0], contact[1], Long.parseLong(contact[2])));
                line = br.readLine();
            }
        } catch (FileNotFoundException e) {
            Interface.fileNotFound();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }

    }

    public static void inputContactsVerticalFormat(String fileName, PhoneBook phoneBook) {
        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
            String line = br.readLine();
            ArrayList<String> contact = new ArrayList<>();
            while (line != null) {
                if (line.length() != 0) {
                    contact.add(line);
                }
                if (contact.toArray().length == 3) {
                    phoneBook.getAllContacts().add(new Contact(contact.get(0), contact.get(1), Long.parseLong(contact.get(2))));
                    contact = new ArrayList<>();
                }
                line = br.readLine();
            }
        } catch (FileNotFoundException e) {
            Interface.fileNotFound();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }

    }

    public static void extractContactsHorisontalFormat(String fileName, PhoneBook phoneBook) throws IOException {
        if (checkFile(fileName)) {
            try {
                PrintWriter pw = new PrintWriter(new File(fileName));
                for (Contact c : phoneBook.getAllContacts()) {
                    pw.write(String.format("%s %s %s\n", c.getName(), c.getSurname(), c.getPhoneNumber()));
                }
                pw.flush();
                pw.close();
            } catch (IOException ex) {
                System.out.println(ex.getMessage());
            }
        }
    }

    public static void extractContactsVerticalFormat(String fileName, PhoneBook phoneBook) throws IOException {
        if (checkFile(fileName)) {
            try {
                PrintWriter pw = new PrintWriter(new File(fileName));
                for (Contact c : phoneBook.getAllContacts()) {
                    pw.write(String.format("%s\n%s\n%s\n\n", c.getName(), c.getSurname(), c.getPhoneNumber()));

                }
                pw.flush();
                pw.close();
            } catch (IOException ex) {
                System.out.println(ex.getMessage());
            }
        }
    }

    public static boolean checkFile(String fileName) throws IOException {
        File outFile = new File(fileName);
        if (!outFile.exists()) {
            if (Interface.fileNotFound()) {
                outFile.createNewFile();
                return true;
            } else {
                return false;
            }
        } else {
            return true;
        }
    }
}

File: homeWorkSeven/python/file_operator.py

from contact import Contact

class FileOperator():

    def __init__(self,interface):
        self.interface = interface

    def input_contacts_horisontal_format(self,file_name:str,phone_book):
        with open(f"{file_name}","r") as file:
            line = file.readline()
            while len(line):
                new_contact = line.split(" ")
                phone_book.get_all_contacts().append(Contact(new_contact[0],new_contact[1],int(new_contact[2])))
                line = file.readline()

    def input_contacts_vertical_format(self,file_name:str,phone_book):
        with open(f"{file_name}","r") as file:
            line = file.readline()
            new_contact = []
            while len(line):
                if len(new_contact) < 4:
                    new_contact.add(line)
                else:
                    phone_book.get_all_contacts().append(Contact(new_contact[0], new_contact[1], int(new_contact[2])))
                    new_contact = []
                line = file.readline()

    def extract_contacts_horisontal_format(self,file_name:str,phone_book):
       if self.check_file(file_name):
           try:
               with open(f"{file_name}","w") as file:
                   for contact in  phone_book.get_all_contacts():
                       file.write(f"{contact.get_name()} {contact.get_surname()} {contact.get_phone_number()}\n")
               self.interface.contact_successfully_extract()
           except:
               print("Ошибка записи!")
       else:
           print("Ошибка записи!")

    def extract_contacts_vertical_format(self,file_name:str,phone_book):
       if self.check_file(file_name):
           try:
               with open(f"{file_name}","w") as file:
                   for contact in  phone_book.get_all_contacts():
                       file.write(f"{contact.get_name()}\n{contact.get_surname()}\n{contact.get_phone_number()}\n\n")
               self.interface.contact_successfully_extract()
           except:
               print("Ошибка записи!")
       else:
           print("Ошибка записи!")

    def check_file(self,file_name:str):
        try:
            file = open(f'{file_name}.txt')
        except IOError as e:
            if self.interface.file_not_found():
                my_file = open(f"{file_name}.txt", "w+")
                my_file.close()
                return True
            else:
                return False
        else:
            return True

if __name__ == '__main__':
    FileOperator().input_contacts_horisontal_format(file_name = "bin_fut")